packages/dcos-launch/extra/launch/onprem.py

```diff
@@ -0,0 +1,138 @@
+import copy
+import logging
+import subprocess
+
+import retrying
+import yaml
+
+import launch.aws
+import launch.util
+
+import test_util.aws
+import test_util.onprem
+import test_util.runner
+from test_util.helpers import Url
+
+log = logging.getLogger(__name__)
+
+STATE_FILE = 'LAST_COMPLETED_STAGE'
+
+
+@retrying.retry(wait_fixed=1000, stop_max_delay=120000)
+def check_ssh(ssher, host, port):
+    ssher.get_home_dir(host, port)
+
+
+class OnpremLauncher(launch.util.AbstractLauncher):
+    def __init__(self, config):
+        # can only be set during the wait command
+        self.bootstrap_host = None
+        self.config = config
+
+    def create(self, config):
+        return self.get_bare_cluster_launcher(config).create(config)
+
+    def post_state(self, state):
+        self.get_ssher(self.config).command(self.bootstrap_host, ['printf', state, '>', STATE_FILE])
+
+    def get_last_state(self):
+        return self.get_ssher(self.config).command(self.bootstrap_host, ['cat', STATE_FILE]).decode().strip()
+
+    def get_bare_cluster_launcher(self, info):
+        if info['platform'] == 'aws':
```

Comment:
No need to validate again, nor would we want to validate in this part of the code. The gist is that we want as much validation as possible to occur when the user providers the config to the CLI (or, more specifically whenever the get validated config function is called). Then, we can leave out distracting validation code and have only the bare business logic which can assume that the config can only be valid.